dart/lib/src/hub.dart

```diff
@@ -309,6 +337,146 @@ class Hub {
       }
     }
   }
+
+  ISentrySpan startTransaction(
+    String name,
+    String operation, {
+    String? description,
+    bool? bindToScope,
+    Map<String, dynamic>? customSamplingContext,
+  }) =>
+      startTransactionWithContext(
+        SentryTransactionContext(
+          name,
+          operation,
+          description: description,
+        ),
+        bindToScope: bindToScope,
+        customSamplingContext: customSamplingContext,
+      );
+
+  ISentrySpan startTransactionWithContext(
+    SentryTransactionContext transactionContext, {
+    Map<String, dynamic>? customSamplingContext,
+    bool? bindToScope,
+  }) {
+    if (!_isEnabled) {
+      _options.logger(
+        SentryLevel.warning,
+        "Instance is disabled and this 'startTransaction' call is a no-op.",
+      );
+    } else if (!_options.isTracingEnabled()) {
+      _options.logger(
+        SentryLevel.info,
+        "Tracing is disabled and this 'startTransaction' returns a no-op.",
+      );
+    } else {
+      final item = _peek();
+
+      final samplingContext = SentrySamplingContext(
+          transactionContext, customSamplingContext ?? {});
+
+      transactionContext.sampled = _tracesSampler.sample(samplingContext);
```

Comment:
we could create a new instance of SentrySamplingContext but this time passing the value of `sampled`